desktop-widgets/divelogexportdialog.cpp

```diff
@@ -229,6 +235,36 @@ void DiveLogExportDialog::export_depths(const char *filename, const bool selecte
 	free_buffer(&buf);
 }
 
+void DiveLogExportDialog::exportProfile(QString filename, const bool selected_only)
+{
+	struct dive *dive;
+	int i;
+	int count = 0;
+	if (!filename.endsWith(".png", Qt::CaseInsensitive))
+		filename = filename.append(".png");
+	QFileInfo fi(filename);
+
+	for_each_dive (i, dive) {
+		if (selected_only && !dive->selected)
+			continue;
+		if (count)
+			saveProfile(dive, fi.path() + "/" + fi.completeBaseName().append(QString("-%1.").arg(count)) + fi.suffix());
```

Comment:
Windows will probably work with the `/` here, but we might want to use `QDir::separator()`